gcs/src/test/java/com/google/cloud/hadoop/fs/gcs/HadoopFileSystemTestBase.java

```diff
@@ -275,6 +275,76 @@ protected void validateListFileInfo(
     }
   }
 
+  @Test
+  public void testInputStreamReadStorageStatistics() throws IOException {
+
+    // Write an object.
+    URI path = GoogleCloudStorageFileSystemIntegrationTest.getTempFilePath();
+    Path hadoopPath = ghfsHelper.castAsHadoopPath(path);
+    String text = "Hello World!";
+    int numBytesWritten = ghfsHelper.writeFile(hadoopPath, text, 1, /* overwrite= */ false);
+    GhfsStorageStatistics StorageStats =
+        new GhfsStorageStatistics(((GoogleHadoopFileSystem) ghfs).getIOStatistics());
+    try (FSDataInputStream readStream = ghfs.open(hadoopPath)) {
+
+      //  Check the statistics of method read().
+      readStream.read();
+      assertThat(StorageStats.isTracked("stream_read_operations")).isTrue();
+      assertThat(StorageStats.getLong("stream_read_operations")).isEqualTo(2);
```

Comment:
why do we have two calls after first read ?

Reply:
Updated